packages/components/src/toggle-group-control/toggle-group-control/as-button-group.tsx

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import type { ForwardedRef } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	useMergeRefs,
+	useInstanceId,
+	usePrevious,
+	useResizeObserver,
+} from '@wordpress/compose';
+import { forwardRef, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { View } from '../../view';
+import ToggleGroupControlBackdrop from './toggle-group-control-backdrop';
+import ToggleGroupControlContext from '../context';
+import { useUpdateEffect } from '../../utils/hooks';
+import type { WordPressComponentProps } from '../../ui/context';
+import type { ToggleGroupControlInnerGroupProps } from '../types';
+import { HStack } from '../../h-stack';
+
+function UnforwardedToggleGroupControlAsButtonGroup(
+	{
+		children,
+		isAdaptiveWidth,
+		label,
+		onChange,
+		size,
+		value,
+		...otherProps
+	}: WordPressComponentProps<
+		ToggleGroupControlInnerGroupProps,
+		'div',
+		false
+	>,
+	forwardedRef: ForwardedRef< HTMLDivElement >
+) {
+	const containerRef = useRef();
+	const [ resizeListener, sizes ] = useResizeObserver();
+	const baseId = useInstanceId(
+		ToggleGroupControlAsButtonGroup,
+		'toggle-group-control-as-button-group'
+	).toString();
+	const [ selectedValue, setSelectedValue ] = useState( value );
+	const groupContext = {
+		baseId,
+		state: selectedValue,
+		setState: setSelectedValue,
+	};
+	const previousValue = usePrevious( value );
+
+	// Propagate groupContext.state change.
+	useUpdateEffect( () => {
+		// Avoid calling onChange if groupContext state changed
+		// from incoming value.
+		if ( previousValue !== groupContext.state ) {
+			onChange( groupContext.state );
+		}
+	}, [ groupContext.state ] );
+
+	// Sync incoming value with groupContext.state.
+	useUpdateEffect( () => {
+		if ( value !== groupContext.state ) {
+			groupContext.setState( value );
+		}
+	}, [ value ] );
+
+	return (
+		<ToggleGroupControlContext.Provider
+			value={ {
+				...groupContext,
+				isBlock: ! isAdaptiveWidth,
+				isDeselectable: true,
+				size,
+			} }
+		>
+			<View
+				aria-label={ label }
+				{ ...otherProps }
+				ref={ useMergeRefs( [ containerRef, forwardedRef ] ) }
+				role="group"
+			>
+				{ resizeListener }
+				<ToggleGroupControlBackdrop
+					state={ groupContext.state }
+					containerRef={ containerRef }
+					containerWidth={ sizes.width }
+					isAdaptiveWidth={ isAdaptiveWidth }
+				/>
+				<HStack spacing={ 1 }>{ children }</HStack>
+			</View>
```

Comment:
The recommended way to group buttons seems to be to [use `section` + `aria-label`](https://twitter.com/LeonieWatson/status/1545745923011117057?ref_src=twsrc%5Etfw%7Ctwcamp%5Etweetembed%7Ctwterm%5E1545745923011117057%7Ctwgr%5E9ea3dd0003e3c99348dfc5550b382c96661f069b%7Ctwcon%5Es1_&ref_url=https%3A%2F%2Flea.verou.me%2F2022%2F07%2Fbutton-group%2F) or [`fieldset` + `legend`](https://twitter.com/simevidas/status/1556994959642869760)

Reply:
In our case I think it would be better to stick to `group` because a lone ToggleGroupControl is likely never going to be worthy of an ARIA landmark in the Gutenberg app. And `fieldset` we need to avoid because of the nesting issue. Any consumer who wants a landmark or a fieldset could do this outside of the component.